src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Tooltip/BoundAttributeDescriptionInfo.cs

```diff
@@ -122,23 +76,35 @@ internal static string ResolveTagHelperTypeName(string returnTypeName, string pr
         // to break it apart and then reconstruct it in a reduced format.
         // i.e. this is the format the display name comes in:
         // ReturnTypeName SomeTypeName.SomePropertyName
+        //
+        // See DefaultBoundAttributeDescriptorBuilder.GetDisplayName() for added detail.
 
-        // We must simplify the return type name before using it to determine the type name prefix
-        // because that is how the display name was originally built (a little hacky).
-        if (!TypeNameStringResolver.TryGetSimpleName(returnTypeName, out var simpleReturnType))
+        var displayNameSpan = displayName.AsSpanOrDefault();
+
+        // Search for the first space, which should be immediately after the return type.
+        var spaceIndex = displayNameSpan.IndexOf(' ');
+        if (spaceIndex < 0)
         {
-            simpleReturnType = returnTypeName;
+            return string.Empty;
         }
 
-        // "ReturnTypeName "
-        var typeNamePrefixLength = simpleReturnType.Length + 1 /* space */;
+        // Increment by one to skip over the space.
+        displayNameSpan = displayNameSpan[(spaceIndex + 1)..];
+
+        var propertyNameSpan = propertyName.AsSpanOrDefault();
 
-        // ".SomePropertyName"
-        var typeNameSuffixLength = /* . */ 1 + propertyName.Length;
+        // Strip off the trailing property name.
+        if (displayNameSpan.EndsWith(propertyNameSpan, StringComparison.Ordinal))
```

Comment:
I considered that and decided that it was fine to just be accurate, since we already have the property name in hand. Note that it's entirely possible for the property name to be null, which means that just checking for a dot might shave off an identifier we shouldn't.